utils/build.ps1

```diff
@@ -696,12 +813,40 @@ function Build-CMakeProject {
       TryAdd-KeyValue $Defines CMAKE_SYSTEM_PROCESSOR $Arch.CMakeName
     }
 
+    if ($Platform -eq "Android") {
+      $vsWherePath = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe"
+      $vsInstallPath = & $vsWherePath -latest -property installationPath
+      if (Test-Path "${vsInstallPath}\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin") {
+        $env:Path = "${vsInstallPath}\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin;${vsInstallPath}\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja;${env:Path}"
+        TryAdd-KeyValue $Defines CMAKE_MAKE_PROGRAM "${vsInstallPath}\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja\ninja.exe"
+      } else {
+        throw "Missing CMake and Ninja in the visual studio installation that are needed to build Android"
+      }
+      $androidNDKPath = Get-AndroidNDKPath
+      TryAdd-KeyValue $Defines CMAKE_C_COMPILER (Join-Path -Path $androidNDKPath -ChildPath "toolchains\llvm\prebuilt\windows-x86_64\bin\clang.exe")
+      TryAdd-KeyValue $Defines CMAKE_CXX_COMPILER (Join-Path -Path $androidNDKPath -ChildPath "toolchains\llvm\prebuilt\windows-x86_64\bin\clang++.exe")
+      TryAdd-KeyValue $Defines CMAKE_ANDROID_API "$AndroidAPILevel"
+      TryAdd-KeyValue $Defines CMAKE_ANDROID_ARCH_ABI $Arch.AndroidArchABI
+      TryAdd-KeyValue $Defines CMAKE_ANDROID_NDK "$androidNDKPath"
+      TryAdd-KeyValue $Defines SWIFT_ANDROID_NDK_PATH "$androidNDKPath"
+      TryAdd-KeyValue $Defines CMAKE_C_COMPILER_WORKS YES
+      TryAdd-KeyValue $Defines CMAKE_CXX_COMPILER_WORKS YES
+      # The current Android NDK ships with Clang 17,
+      # which doesn't provide the _Builtin_float module.
+      TryAdd-KeyValue $Defines SWIFT_BUILD_CLANG_OVERLAYS_SKIP_BUILTIN_FLOAT YES
+    }
+
     TryAdd-KeyValue $Defines CMAKE_BUILD_TYPE Release
     TryAdd-KeyValue $Defines CMAKE_MT "mt"
 
     $CFlags = @()
-    if ($Platform -eq "Windows") {
-      $CFlags = @("/GS-", "/Gw", "/Gy", "/Oi", "/Oy", "/Zc:inline")
+    switch ($Platform) {
+      Windows {
+        $CFlags = @("/GS-", "/Gw", "/Gy", "/Oi", "/Oy", "/Zc:inline")
+      }
+      Android {
+        $CFlags = @("--sysroot=$(Get-AndroidNDKPath)\toolchains\llvm\prebuilt\windows-x86_64\sysroot")
```

Comment:
If this means that we don't support the non-x86_64 build/host machine such as windows arm64, could we check/assert that here?

Reply:
Unfortunately, the NDK does not have the ARM64 slice of the toolchain, so we have to assert here :(.